### ISSUE TITLE
Check if high speed camera is enabled

### ISSUE THREAD

**Issue by alexpiet** [AllenNeuralDynamics/dynamic-foraging-task#190]
- Currently in the training room, there is no high speed camera. In the ephys rig there is.
- We need to modify the bonsai workflow to gracefully check if a high speed camera is enabled or not. If a camera is present, enable the node. If a camera is not present, disable the node. 
- this is required before automatic updates in the ephys rigs

**Comment by alexpiet** [comment]
@bruno-f-cruz Can you point us in the right direction?

**Comment by bruno-f-cruz** [comment]
I would advise solving this at the level of the schema tbh. If there is a camera entry try to connect if not don't subscribe to the camera. I have shown xin xin how to do this with the nidaq already. Use the same pattern. Otherwise you could simply try to connect to the camera and if it doesn't exist catch the error and ignore it. Which one would you prefer?

**Comment by XX-Yin** [comment]
Hi @bruno-f-cruz, do you have time to show me the `catch the error and ignore`? The excel schema method does not work for the nidaq.

**Comment by bruno-f-cruz** [comment]
Pick a time from my next week s calendar. Meanwhile have you tried the docs?

https://bonsai-rx.org/docs/api/Bonsai.Reactive.Catch.html
https://bonsai-rx.org/docs/api/Bonsai.Reactive.OnErrorResumeNext.html

**Comment by alexpiet** [comment]
![IMG_20240321_092033181](https://github.com/AllenNeuralDynamics/dynamic-foraging-task/assets/7605170/a083bed5-ecc6-427b-af1f-c132923dc91a)

@bruno-f-cruz Showed @XX-Yin and I how to do this. The idea is we put a value in the rig `.csv` file, and then set the workflow to only subscribe to the high speed camera node if the appropriate value is in the rig file.

I will update the setting files to contain an entry `HighSpeedCamera`. It will default to `0`, and if its `0`, the camera node will not be subscribed to. If you want to use a high speed camera, then you set the node to `1`

**Comment by bruno-f-cruz** [comment]
For completeness sake:

![image](https://github.com/AllenNeuralDynamics/dynamic-foraging-task/assets/7049351/6cfa5a2d-9b97-4913-8544-4eb55a80c2a2)


<details>
    <summary>Workflow (Copy+Paste me)</summary>
    
```xml

<?xml version="1.0" encoding="utf-8"?>
<WorkflowBuilder Version="2.8.1"
                 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                 xmlns:rx="clr-namespace:Bonsai.Reactive;assembly=Bonsai.Core"
                 xmlns="https://bonsai-rx.org/2018/workflow">
  <Workflow>
    <Nodes>
      <Expression xsi:type="GroupWorkflow">
        <Name>RandomPieceOfHardware</Name>
        <Workflow>
          <Nodes />
          <Edges />
        </Workflow>
      </Expression>
      <Expression xsi:type="Annotation">
        <Name>Any boolean expression</Name>
        <Text><![CDATA[]]></Text>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="BooleanProperty">
          <Value>false</Value>
        </Combinator>
      </Expression>
      <Expression xsi:type="rx:Condition">
        <Workflow>
          <Nodes>
            <Expression xsi:type="WorkflowInput">
              <Name>Source1</Name>
            </Expression>
            <Expression xsi:type="WorkflowOutput" />
          </Nodes>
          <Edges>
            <Edge From="0" To="1" Label="Source1" />
          </Edges>
        </Workflow>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="rx:SubscribeWhen" />
      </Expression>
    </Nodes>
    <Edges>
      <Edge From="0" To="4" Label="Source1" />
      <Edge From="1" To="2" Label="Source1" />
      <Edge From="2" To="3" Label="Source1" />
      <Edge From="3" To="4" Label="Source2" />
    </Edges>
  </Workflow>
</WorkflowBuilder>

```

</details>

**Comment by XX-Yin** [comment]
@alexpiet I can make a PR to update this and also read the camera's serial number from the CSV file.

**Comment by alexpiet** [comment]
Resolved by #318
